src/services/BookingService.ts

```diff
@@ -46,14 +51,39 @@ class BookingService {
 
 
     if (bookingsAmount !== 0) {
-      throw new HttpException();
+      throw new HttpException(
+        500, "Slot is already occupied.",
+      );
     }
 
 
     return false;
   }
 
+  public async doesTheBookingOfTheUserOverlap(
+    userId: number,
+    startDate: string,
+    endDate: string,
+  ): Promise<boolean> {
+    const repo = this.getRepository();
+
+    const userBookings = await repo.findByIdAndDates(
```

Comment:
Please make this method specifically for this use case. It's better to have to use case-specific functionalities that are easier to maintain and have a single responsibility (either querying based on slots, or query based on users) that have to maintain two different behaviors which may change for different reasons (if we have some change for slots, then we'd need to add an `if` statement on this method only for checking `if entity === "slot"`).